ReSharper.FSharp/src/FSharp.Psi.Intentions/src/Intentions/UseNamedAccessAction.fs

```diff
@@ -0,0 +1,103 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Intentions
+
+open FSharp.Compiler.Symbols
+open JetBrains.ReSharper.Feature.Services.ContextActions
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open type JetBrains.Diagnostics.Assertion
+
+[<ContextAction(Name = "UseNamedAccess", Group = "F#", Description = "Use named access inside a DU pattern")>]
+type UseNamedAccessAction(dataProvider: FSharpContextActionDataProvider) =
+    inherit FSharpContextActionBase(dataProvider)
+    
+    let mutable names = Array.empty
+    let mutable tuplePat : ITuplePat = null
+
+    override x.Text = "Use named access"
+
+    override x.IsAvailable _ =
+        let pattern = dataProvider.GetSelectedElement<IParametersOwnerPat>()
+        if isNull pattern then false else
+        // We expect a single IParenPat here
+        if pattern.Parameters.Count <> 1 then false else
+        let parenPat =
+            match pattern.Parameters.[0] with
+            | :? IParenPat as parenPat -> parenPat
+            | _ -> null
+
+        if isNull parenPat then false else
+
+        // Ignore multiline patterns for now
+        if parenPat.StartLine <> parenPat.EndLine then false else
+
+        // This should only work when there are multiple items
+        tuplePat <- parenPat.Pattern :?> ITuplePat
+        if isNull tuplePat then false else
+
+        // We need to be sure that the pattern is a DU with all named fields
+        let fcsUnionCase = pattern.ReferenceName.Reference.GetFcsSymbol().As<FSharpUnionCase>()
```

Comment:
In theory `pattern.ReferenceName` might be `null` in case of some broken code.

The reference name may also be somewhat broken, but we shouldn't fail in that case too. Please add a test case showing that the context action isn't available here (meaning there're no exceptions produced):
```fsharp
match x with
| Namespace. (a, b) -> ()
```

Reply:
`IParametersOwnerPat` is already null in this example.

Reply:
```
Language: PsiLanguageType:F#
IFSharpImplFile
  IAnonModuleDeclaration
    IExpressionStatement
      IChameleonExpression
        IMatchExpr
          FSharpTokenType+MatchTokenElement(type:MATCH, text:match)
          Whitespace(type:WHITE_SPACE, text: ) spaces:" "
          IReferenceExpr
            FSharpIdentifierToken(type:IDENTIFIER, text:x)
          Whitespace(type:WHITE_SPACE, text: ) spaces:" "
          FSharpTokenType+WithTokenElement(type:WITH, text:with)
          NewLine(type:NEW_LINE, text:\n) spaces:"\n"
          IMatchClause
            FSharpTokenType+BarTokenElement(type:BAR, text:|)
            Whitespace(type:WHITE_SPACE, text: ) spaces:" "
            ILocalReferencePat
              IExpressionReferenceName
                IExpressionReferenceName
                  FSharpIdentifierToken(type:IDENTIFIER, text:Namespace)
                FSharpTokenType+DotTokenElement(type:DOT, text:.)
                Whitespace(type:WHITE_SPACE, text: ) spaces:" "
                FSharpTokenType+LparenTokenElement(type:LPAREN, text:()
                FSharpIdentifierToken(type:IDENTIFIER, text:a)
                FSharpTokenType+CommaTokenElement(type:COMMA, text:,)
                Whitespace(type:WHITE_SPACE, text: ) spaces:" "
                FSharpIdentifierToken(type:IDENTIFIER, text:b)
                FSharpTokenType+RparenTokenElement(type:RPAREN, text:))
            Whitespace(type:WHITE_SPACE, text: ) spaces:" "
            FSharpTokenType+RarrowTokenElement(type:RARROW, text:->)
            Whitespace(type:WHITE_SPACE, text: ) spaces:" "
            IUnitExpr
              FSharpTokenType+LparenTokenElement(type:LPAREN, text:()
              FSharpTokenType+RparenTokenElement(type:RPAREN, text:))
              ```

Reply:
> IParametersOwnerPat is already null in this example.

I can see it changing, though. 🙂

Reply:
Oh, I did add the check for `pattern.ReferenceName` being `null``.
The current test just doesn't cover it.